Add Encryption Of Backup Files As An Option

vicky16898 opened the pull request:
This commit allows users to encrypt the backup files when
cipher module is enabled.

hariprasanths commented on README.md:
```diff
@@ -106,6 +105,7 @@ Initializing configurations can also be done using a file, `synchly --config=mod
 By default, remote-sync and smtp modules are disabled, to enable them, use `synchly --enable=module` command.
 
 To encyrpt the job configuration files use the command `synchly --enable=cipher` and to disable it globally use the command `synchly --disable=cipher`.
+Encryption of backup files can be done when cipher module is enabled.
```
change it to only when?

hariprasanths commented on src/database/mongoDb/mongoDb.js:
```diff
@@ -64,6 +76,7 @@ let restore = async (dbConfig, backupFilename) => {
     }
 
     let dbRestore = await exec(mongoRestoreCmd);
+    files.deleteFile(backupFilePath);
```
won't this delete the backup file if encryption is disabled?

hariprasanths commented on src/database/mysql/mysql.js:
```diff
@@ -58,18 +57,32 @@ let dump = async (dbConfig, backupPath) => {
     if (isCompressed) {
-        const decompressFileRes = await files.decompressFile(backupFilePath);
-        backupFilePath = backupFilePath + '.sql';
+        console.log(isCompressed);
```
remove debug logs

hariprasanths commented on src/utils/files.js:
```diff
@@ -75,6 +79,74 @@ const decompressFile = async (filename) => {
+        const readStream = fs.createReadStream(tempFileName);
+        const cipher = crypto.createCipheriv('aes256', CIPHER_KEY, initVect);
+        const appendInitVect = new AppendInitVect(initVect);
+        const writeStream = fs.createWriteStream(path.join(filePath));
```
unnecessary path.join?

hariprasanths commented on src/database/mysql/mysql.js:
```diff
@@ -58,18 +57,32 @@ let dump = async (dbConfig, backupPath) => {
+            await files.encrypt(backupPath, key);
+        }
+    } catch (err) {
+        console.log('Encryption Failed.');
```
log this inside the catch of encrypt

hariprasanths commented on src/utils/files.js:
```diff
@@ -75,6 +79,75 @@ const decompressFile = async (filename) => {
+        return await promise;
+    } catch (err) {
+        deleteFile(filePath);
+        console.log('Encryption Failed.');
```
Remove this or the log on the higher level

vicky16898 committed:
Add Encryption Of Backup Files As An option

This commit allows users to encrypt the backup files
when cipher module is enabled

hariprasanths commented on src/utils/files.js:
```diff
@@ -75,6 +79,78 @@ const decompressFile = async (filename) => {
+    try {
+        fs.renameSync(filePath, tempFileName);
+        const initVect = crypto.randomBytes(16);
+        const encryptionFlag = Buffer.from('BEtr', 'utf-8');
```
Use the string from the strings.js

hariprasanths commented on src/utils/files.js:
```diff
@@ -2,6 +2,10 @@ const fs = require('fs');
 const zlib = require('zlib');
+const crypto = require('crypto');
+const AppendInitVect = require('./appendInitVector');
+const {resolve} = require('path');
```
remove unused imports

hariprasanths commented on src/database/inquirer.js:
```diff
@@ -119,6 +119,13 @@ let askConfig = async (jobName, key) => {
             }
         },
     });
+    if (key) {
```
Show a log to enable cipher module if its disabled

hariprasanths commented on src/utils/strings.js:
```diff
@@ -178,4 +178,5 @@ module.exports = {
     moduleStatusDisabled: 'disabled',
```
Import constants from constants.js
